ultra/evaluate.py

```diff
@@ -0,0 +1,218 @@
+            [
+                evaluate.remote(
+                    experiment_dir=episode.experiment_dir,
+                    agent_id="AGENT_008",
+                    policy_class=policy_class,
+                    seed=episode.eval_count,
+                    itr_count=agent_itr,
+                    checkpoint_dir=checkpoint_dir,
+                    scenario_info=scenario_info,
+                    num_episodes=eval_episodes,
+                    headless=headless,
+                    timestep_sec=timestep_sec,
+                )
+            ]
+        )[0]
+        episode.eval_count += 1
+        episode.last_eval_iteration = agent_itr
+        episode.record_tensorboard(agent_id=agent_id)
+        episode.train_mode()
+
+
+# Number of GPUs should be splited between remote functions.
+@ray.remote(num_gpus=num_gpus / 2)
+def evaluate(
+    experiment_dir,
+    seed,
+    agent_id,
+    policy_class,
+    itr_count,
+    checkpoint_dir,
+    scenario_info,
+    num_episodes,
+    headless,
+    timestep_sec,
+):
+
+    torch.set_num_threads(1)
+    spec = make(
+        locator=policy_class,
+        checkpoint_dir=checkpoint_dir,
+        experiment_dir=experiment_dir,
+    )
+
+    env = gym.make(
+        "ultra.env:ultra-v0",
+        agent_specs={agent_id: spec},
+        scenario_info=scenario_info,
+        headless=headless,
+        timestep_sec=timestep_sec,
+        seed=seed,
+        eval_mode=True,
+    )
+
+    agent = spec.build_agent()
+    summary_log = LogInfo()
+    logs = []
+
+    for episode in episodes(num_episodes):
+        observations = env.reset()
+        state = observations[agent_id]
+        dones, infos = {"__all__": False}, None
+
+        episode.reset(mode="Evaluation")
+        while not dones["__all__"]:
+            action = agent.act(state, explore=False)
+            observations, rewards, dones, infos = env.step({agent_id: action})
+
+            next_state = observations[agent_id]
+
+            state = next_state
+
+            episode.record_step(agent_id=agent_id, infos=infos, rewards=rewards)
+
+        episode.record_episode()
+        logs.append(episode.info[episode.active_tag].data)
+
+        for key, value in episode.info[episode.active_tag].data.items():
+            if not isinstance(value, (list, tuple, np.ndarray)):
+                summary_log.data[key] += value
+
+    for key, val in summary_log.data.items():
+        if not isinstance(val, (list, tuple, np.ndarray)):
+            summary_log.data[key] /= num_episodes
+
+    env.close()
+
+    return summary_log
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("intersection-single-agent-evaluation")
+    parser.add_argument(
+        "--task", help="Tasks available : [1, 2]", type=str, default="1"
+    )
+    parser.add_argument(
+        "--level",
+        help="Levels available : [easy, medium, hard, no-traffic]",
+        type=str,
+        default="easy",
+    )
```

Comment:
Just thinking, should these be mandatory arguments?
```suggestion
    parser.add_argument(
        "--task", help="Tasks available : [1, 2]", type=str, required=True
    )
    parser.add_argument(
        "--level",
        help="Levels available : [easy, medium, hard, no-traffic]",
        type=str,
        required=True,
    )
```

Reply:
They aren't mandatory since there are defaults. They currently match the defaults in train.py, but maybe we don't want to do that.

Reply:
Actually sorry, for now I think it's probably okay to have default values here; just for the initial ULTRA version.